pkg/ensign/store/meta/names.go

```diff
@@ -200,6 +200,43 @@ func (s *Store) TopicName(topicID ulid.ULID) (_ string, err error) {
 	return topic.Name, nil
 }
 
+// LookupTopicName returns a topicID for the specified topic name and project.
+func (s *Store) LookupTopicName(name string, projectID ulid.ULID) (topicID ulid.ULID, err error) {
```

Comment:
This new method uses the topic names index to perform a unique lookup of the topic ID for checking if a query has access to the specified topic.